.github/workflows/go-tests.yml

```diff
@@ -498,20 +498,36 @@ jobs:
     steps:
       - name: evaluate upstream job results
         run: |
+          NEEDS_JSON='${{ toJSON(needs) }}'
+          
+          # if failure (not cancelled), notify Slack
```

Comment:
**question**: would this be easier to read if it was nested in the below if for failures and cancellations?  Or possibly exporting NEEDS_JSON and including this logic in the `Set Slack commit message summary` job instead?

Reply:
Fair questions! I initially preferred keeping the near-identical failure detection close together to discourage future drift - but I think these are simple and close enough already to make it fair to split them. I'll move the Slack check down 👍🏻 

I was also 50/50 on the value of `NEEDS_JSON` to begin w/, and cutting that would probably aid legibility at the expense of a tiny bit more copypasta, so I might skip on the export for now and just go back to inlining the parse.

Reply:
Ah, actually I just remembered why I did it this way - been several iterations since last week 😄 

* If we put Slack second, but detect failure in that step, a cancellation -> `exit 1` in the summary check will cause the failure-only check to fail.
* If we put Slack first, and it fails for some superfluous reason due to a timeout/etc., it will cause the summary check to fail.

So I think keeping them together is probably ideal.

I think nesting one check inside another could hurt legibility, but inlining `NEEDS_JSON` feels like it might help a bit, so I'll at least do that.

Reply:
Done! I also swapped the `always()` for `failure()` checks to avoid needless steps on success.

Re-tested:
- Cancelled: https://github.com/hashicorp/consul/actions/runs/7090874235/job/19298822105
- Failure: https://github.com/hashicorp/consul/actions/runs/7090874235/job/19298858923#step:4:11
- Success: https://github.com/hashicorp/consul/actions/runs/7090908496/job/19298936417